exchange/bidder.go

```diff
@@ -295,6 +296,10 @@ func (bidder *bidderAdapter) doRequest(ctx context.Context, req *adapters.Reques
 	if err != nil {
 		if err == context.DeadlineExceeded {
 			err = &errortypes.Timeout{Message: err.Error()}
+			if tb, ok := bidder.Bidder.(adapters.TimeoutBidder); ok {
+				go bidder.doTimeoutNotification(tb, req)
```

Comment:
This might benefit from a comment explaining the purposeful 'set it and forget it' approach. Should be also be concerned about creating too many go routines? Probably not, but wanted to seek opinions.

Reply:
We can only do one timeout call per timeout, so it is probably not too much of an issue.

Reply:
I'm split about proposing this because it does imply having an extra function to maintain and the benefits are marginal to say the least but, do you think a function to check whether or not the bidder implements `MakeTimeoutNotification` makes the code a bit easier to read?
```
283   // doRequest makes a request, handles the response, and returns the data needed by the
284   // Bidder interface.
285   func (bidder *bidderAdapter) doRequest(ctx context.Context, req *adapters.RequestData) *httpCallInfo {
286       httpReq, err := http.NewRequest(req.Method, req.Uri, bytes.NewBuffer(req.Body))
287   +--  9 lines: if err != nil {---------------------------------------------------------------------------
296       if err != nil {
297           if err == context.DeadlineExceeded {
298               err = &errortypes.Timeout{Message: err.Error()}
299 -             if tb, ok := bidder.Bidder.(adapters.TimeoutBidder); ok {
    +             if tb, ok := isTimeoutImplemented(bidder.Bidder); ok {
300                   go bidder.doTimeoutNotification(tb, req)
301               }
302  
303           }
304           return &httpCallInfo{
305               request: req,
306               err:     err,
307           }
308       }
309   +-- 16 lines: respBody, err := ioutil.ReadAll(httpResp.Body)--------------------------------------------
310       return &httpCallInfo{
326           request: req,
327           response: &adapters.ResponseData{
328               StatusCode: httpResp.StatusCode,
329               Body:       respBody,
330               Headers:    httpResp.Header,
331           },
332           err: err,
333       }
334   }
335  
    + func isTimeoutImplemented(adapterBidder adapters.Bidder) (adapters.TimeoutBidder, bool) {
    +     return adapterBidder.(adapters.TimeoutBidder)
    + }
     
exchange/bidder.go [+]
```